azuredevops/internal/service/serviceendpoint/resource_serviceendpoint_gcp.go

```diff
@@ -0,0 +1,83 @@
+package serviceendpoint
+
+import (
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/v6/serviceendpoint"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/converter"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/internal/utils/tfhelper"
+)
+
+// ResourceServiceEndpointGcp schema and implementation for gcp service endpoint resource
+func ResourceServiceEndpointGcp() *schema.Resource {
+	r := genBaseServiceEndpointResource(flattenServiceEndpointGcp, expandServiceEndpointGcp)
+	r.Schema["client_email"] = &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    false,
```

Comment:
if this property is required,  use `Required:    true,` and  if optional use `Optional:    true,`